### PR TITLE
Fix NTSC Designer

### PR DESCRIPTION
**Pull request by aldoir**

### CODE REVIEW

File: Tools/OSD/osd_screen.cs [hunk 1]
```diff
@@ -604,12 +606,6 @@ private void rbtSortAlphabetic_CheckedChanged(object sender, EventArgs e)
 			}
 		}
 		
-        private int correct_NTSC(int y){
-            if (y >= osd.getCenter() && osd.is_ntsc()) 
-                y+=3;
-            return y;
-        }
-
```
Comment (night-ghost) [inline Tools/OSD/osd_screen.cs]: what has prevented this feature? Occam's Razor - it is not necessary to produce the spirit, it is not necessary to embed checking code in every place! It must to split out to a function

File: Tools/OSD/osd_screen.cs [hunk 2]
```diff
@@ -625,28 +621,45 @@ private void rbtSortAlphabetic_CheckedChanged(object sender, EventArgs e)
...
 
 
-        private void setYpos(int y) {
-            string item;
-
-            item = osd.currentlyselected;
-
-            for (int a = 0; a < panelItems.Length; a++) {
-                if (panelItems[a] != null && panelItems[a].name == item) {
-                    panelItems[a].y = y;
```
Comment (night-ghost) [inline Tools/OSD/osd_screen.cs]: and the same thing here

File: Tools/OSD/osd_screen.cs (hunk 1 above)
Comment (aldoir) [inline Tools/OSD/osd_screen.cs]: This function adds 3 lines to the object every time you click on a node item.

I'm going to refactor the whole class to be OOP, don't worry about this

Comment (aldoir) [inline Tools/OSD/osd_screen.cs]: BTW, it's not about checking. There are 2 different things, one is the drag&drop and other is the up/down control. Unfortunatelly I couldn't reuse the same logic for both situations.

Comment (night-ghost) [inline Tools/OSD/osd_screen.cs]: >This function adds 3 lines to the object every time you click on a node item.

we should not worry about the size and speed of the CT, the only requirement - it should be clear

Comment (night-ghost) [inline Tools/OSD/osd_screen.cs]: >I'm going to refactor the whole class to be OOP

NEVER

Comment (aldoir) [inline Tools/OSD/osd_screen.cs]: Ok, then do it in your way (Or not). And keep being sarcastic that PRs are welcome

Comment (night-ghost) [inline Tools/OSD/osd_screen.cs]: yes PRs are welcome - but not "I will bear everything and remake in my own way". Always some rules exists. Do you try to make PR to Ardupilot or Linux kernel?

Comment (aldoir) [inline Tools/OSD/osd_screen.cs]: Well, the fix is there,  use if you want, otherwise discard it.

**Comment by aldoir** [comment]
This time tried to be less noisy on the changes. Unfortunatelly can't be done in one playce only

File: Tools/OSD/osd_screen.cs [hunk 3]
```diff
@@ -656,20 +665,34 @@ private void rbtSortAlphabetic_CheckedChanged(object sender, EventArgs e)
...
+                {
+                    // started on top screen and went to bottom
+                    if (topScreen && ansH >= centerNTSC)
+                    {
+                        ansH += 3;
+                    }
+                    // started on bottom screen and went to top
+                    else if (!topScreen && ansH <= centerPAL)
+                    {
+                        ansH -= 3;
```
Comment (night-ghost) [inline Tools/OSD/osd_screen.cs]: Why to change something in the addressing of lines in the PAL mode?

File: Tools/OSD/osd_screen.cs [hunk 4]
```diff
@@ -656,20 +665,34 @@ private void rbtSortAlphabetic_CheckedChanged(object sender, EventArgs e)
...
+                int centerNTSC = (OSD.SCREEN_H_NTSC / 2);
+                getCharLoc(e.X, e.Y, out ansW, out ansH);
 				//if(ansH >= osd.getCenter() && !(osd.pal_checked() || osd.auto_checked())) {
 					//ansH += 3;
 				//}
 				ansW -= clickX; //запомним куда ткнули
-				ansH -= clickY;
-				                
-				NUM_X.Value = Constrain(ansW, 0, osd.get_basesize().Width - 1);
-                NUM_Y.Value = Constrain(ansH, 0, OSD.SCREEN_H - 1);
```
Comment (night-ghost) [inline Tools/OSD/osd_screen.cs]: why you remove Constrain()?

File: Tools/OSD/osd_screen.cs [hunk 5]
```diff
@@ -690,8 +713,17 @@ private void rbtSortAlphabetic_CheckedChanged(object sender, EventArgs e)
         private void pictureBox1_MouseDown (object sender, MouseEventArgs e) {
 			osd.BeginInvoke((MethodInvoker)delegate {
 				osd.currentlyselected = getMouseOverItem(e.X, e.Y);
-                adjustGroupbox();
-			});
+                adjustGroupbox();
```
Comment (night-ghost) [inline Tools/OSD/osd_screen.cs]: Is working with the control of of another thread should not be done through the Invoke?

File: Tools/OSD/osd_screen.cs (hunk 4 above)
Comment (aldoir) [inline Tools/OSD/osd_screen.cs]: I didn't, check the raw file, I think github is crazy about indentation

File: Tools/OSD/osd_screen.cs (hunk 3 above)
Comment (aldoir) [inline Tools/OSD/osd_screen.cs]: I need to skip lines 6,7,8 (center of NTSC to center of PAL). Didn't want to hardcode it. Any suggestion?

File: Tools/OSD/osd_screen.cs (hunk 5 above)
Comment (aldoir) [inline Tools/OSD/osd_screen.cs]: I kept it inside the delegate block, but it's not necessary as it wont deal with any UI control.

File: Tools/OSD/osd_screen.cs (hunk 4 above)
Comment (aldoir) [inline Tools/OSD/osd_screen.cs]: I did check the file and there is a mix between tabs and 4 spaces as indentation. Which should I use?

By default VS uses 4 spaces

Comment (night-ghost) [inline Tools/OSD/osd_screen.cs]: without a difference, I do not use GIT - I use a "diff -Bbu"

File: Tools/OSD/osd_screen.cs (hunk 3 above)
Comment (night-ghost) [inline Tools/OSD/osd_screen.cs]: do not interfere with the work in PAL mode

File: Tools/OSD/osd_screen.cs (hunk 5 above)
Comment (night-ghost) [inline Tools/OSD/osd_screen.cs]: now yes don't deal but this function is indendent to do it.

File: Tools/OSD/osd_screen.cs (hunk 3 above)
Comment (aldoir) [inline Tools/OSD/osd_screen.cs]: did you see everything is inside a if (osd.is_ntsc()) block? What's your suggestion?

**Comment by night-ghost** [comment]
Ok

**Pull request merged**